src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectFactory.cs

```diff
@@ -164,6 +163,10 @@ await ApplyChangeToWorkspaceAsync(w =>
                 onAfterUpdate: null);
         }).ConfigureAwait(false);
 
+        // Set this value early after solution is created so it is available to Razor.  This will get updated
+        // when the command line is set, but we want a non-null value to be available as soon as possible.
+        project.CompilationOutputAssemblyFilePath = creationInfo.CompilationOutputAssemblyFilePath;
```

Comment:
Could this have essentially been the old change needed here?

```
_projectUpdateState = _projectUpdateState.WithProjectOutputPath(creationInfo.CompilationOutputAssemblyFilePath, creationInfo.ProjectId);
```

Reply:
Possibly - but going through the same property setter as everything else feels more safer.